compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -424,6 +433,11 @@ fn report_arm_reachability<'p, 'tcx>(
                     }
 
                     hir::MatchSource::ForLoopDesugar | hir::MatchSource::Normal => {
+                        if matches!(report.reachable,
+                            NonExhaustiveReachable::MayLint { .. } if matches!(arm.pat.kind.as_ref(), PatKind::Wild)
```

Comment:
This prevents the `unreachable_patterns` lint from being triggered since with a `non_exhaustive` enum I have removed the unmatchable `Constructor::NonExhaustive` from the matrix/pat stack it thinks the `_` wildcard match is unreachable. This basically says if we have altered the matrix (now the `UsefulnessReport`) because of `non_exhaustive_reachable_patterns` lint don't emit `unreachable_patterns` lint.

I think this is ok, there is no case where a wildcard for a `non_exhaustive` enum is unreachable right?